hazelcast-cloud/src/main/java/com/hazelcast/aws/utility/CloudyUtility.java

```diff
@@ -43,19 +43,6 @@
     private CloudyUtility() {
     }
 
-    public static String getQueryString(Map<String, String> attributes) {
-        StringBuilder query = new StringBuilder();
-        for (final Map.Entry<String, String> entry : attributes.entrySet()) {
-            final String value = entry.getValue();
-            query.append(AwsURLEncoder.urlEncode(entry.getKey())).append("=").append(AwsURLEncoder.urlEncode(value)).append("&");
-        }
-        String result = query.toString();
-        if (result != null && !result.equals("")) {
-            result = "?" + result.substring(0, result.length() - 1);
-        }
-        return result;
-    }
-
     public static Map<String, String> unmarshalTheResponse(InputStream stream, AwsConfig awsConfig) throws IOException {
         return parseAddresses(stream, awsConfig);
```

Comment:
Just copy the content of parseAddresses into the unmarshallTheResponse method.  Since the parseAddresses is not used in any other place and the unmarshalTheResponse is not adding any extra logic, it is best to get rid of complexity.